iotile_ext_cloud/test/conftest.py

```diff
@@ -103,6 +104,9 @@ def ota_cloud(mock_cloud_private_nossl):
 @pytest.fixture(scope="function")
 def simple_hw():
 
+    if sys.version_info < (3, 5):
```

Comment:
Any advantage in supporting 3.5? In my experience, people (e.g. AWS) seem to have jumped from 3.4 all the way to 3.6. If we are moving to Python3, I think we should just go all the way to 3.6.

Reply:
I'm onboard with running CoreTools on python 3.6 by default and encouraging users to always use the most recent stable interpreter but there don't seem to be any major changes between 3.5 and 3.6 that prevents us from getting 3.5 compatibility for free.  The only major advantage seems to be the presence of `f-strings` on python 3.6 to remove the need to call `str.format()` for simple string templating.  

When we go fully over to python 3 only there may be advantages in the better `typing` module for type annotations but other than that it seems like the major difference is some syntactic sugar for asynchronous list comprehensions:
https://docs.python.org/3/whatsnew/3.6.html

Supporting python 3.4 seemed like a burden because 3.5 adds `async/await` and is a huge step up in terms of writing coroutines but moving from 3.5 to 3.6 seems somewhat incremental.  Until we see a clear benefit to dropping python 3.5 entirely, I would rather default to supporting it.  Dropping support for python 3.4 was important to allow writing true coroutines with `async def`:

https://docs.python.org/3/whatsnew/3.6.html

Reply:
There are a few niceties of py3.6 with async/await that may just be nice to have, based on that what's new page, such as https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep525 (allowing await and yield in the same function body), as well as async generators. 

I agree that there aren't any game changers, but 3.6 gives us some good performance boosts on asyncio, because technically on 3.5 it's a provisional feature.


That being said, it's also less of a break to ask that users upgrade from 3.5 to 3.6, compared to asking for a move from 2.7 to a new major release, if we do change our minds later.

Reply:
Although, if we continue down that path, 3.7 *also* significantly enhances asyncio: 
https://docs.python.org/3/whatsnew/3.7.html#whatsnew37-asyncio